### PR TITLE
Support for COPY INTO

### LINKED ISSUE #41: Support for COPY INTO
### Describe the feature
Databricks support the ability to load data into tables using [COPY INTO](https://docs.databricks.com/spark/latest/spark-sql/language-manual/delta-copy-into.html).

### Describe alternatives you've considered
I can write my own macro

### Who will this benefit?
A typical use case is to load files from cloud storage into a 'bronze' Delta table.

### Are you interested in contributing this feature?
sure!

### PR DESCRIPTION
**Pull request by allisonwang-db**
resolves #41

### Description
This PR adds a macro `databricks_copy_into` to support the [COPY INTO](https://docs.databricks.com/spark/latest/spark-sql/language-manual/delta-copy-into.html) command. You can use it with dbt run-operation:
```bash
dbt run-operation databricks_copy_into --args ...
```

### Checklist

- [ ] I have run this code in development and it appears to resolve the stated issue
- [ ] This PR includes tests, or tests are not required/relevant for this PR
- [ ] I have updated the `CHANGELOG.md` and added information about my change to the "dbt-databricks next" section.

### CODE REVIEW

File: tests/integration/copy_into/test_copy_into.py [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+        self.run_dbt(["run-operation", "databricks_copy_into", "--args", args.format(source_path=path)])
+        self.assertTablesEqual("target", "expected_target")
+
+    @use_profile("databricks_cluster")
+    def test_databricks_cluster(self):
+        self.test_copy_into()
+
+    @use_profile("databricks_sql_endpoint")
+    def test_databricks_sql_endpoint(self):
+        self.test_copy_into()
```
Comment (ueshin) [inline tests/integration/copy_into/test_copy_into.py:43]: Does this feature not support UC?

**Review by ueshin — APPROVED** [review]
@allisonwang-db Thanks for working on this!

Could you fix the linter issue by running `tox -e black`?
Could you also add an item in `CHANGELOG.txt` for this?

Otherwise, LGTM.

File: tests/integration/copy_into/test_copy_into.py (hunk 1 above)
Comment (allisonwang-db) [reply tests/integration/copy_into/test_copy_into.py:43]: I tested against a UC cluster and I don't think UC can support this test case:
1. UC does not allow the creation of parquet tables (only Delta is supported)
2. UC does not allow access to the underlying path of a managed Delta table (so we can't get the source in the test)

But COPY INTO should support UC in general: https://docs.databricks.com/ingestion/copy-into/unity-catalog.html

**Comment by ueshin** [comment]
Thanks! merging.

**Comment by bilalaslamseattle** [comment]
@ueshin  does this support the [inline credentials](https://docs.databricks.com/spark/latest/spark-sql/language-manual/delta-copy-into.html) that newer DBRs support for COPY INTO?

**Comment by ueshin** [comment]
@bilalaslamseattle Yes, seems so, if you mean this:

https://github.com/databricks/dbt-databricks/blob/1079d25df6b4c68561a3e5636daa734278b2112e/dbt/include/databricks/macros/copy_into.sql#L27-L33

cc @allisonwang-db

**Comment by allisonwang-db** [comment]
@bilalaslamseattle Yes, it should support the inline credentials (for the source table).

**Pull request merged**